### DISCUSSION TITLE
Clonar seu próprio projeto do GitHub em uma máquina diferente

### DISCUSSION THREAD

**Discussion by ValdirCezar** [frontendbr/forum#1929] [Perguntas e Respostas] [answered]
Olá, pessoal!

Estou iniciando os estudo com Git e GitHub e tenho uma dúvida que não consegui encontrar na documentação ou em fóruns.

1 -> Eu tenho 2 computadores, o pessoal e o de trabalho.
2 -> Iniciei um projeto no GitHub com a máquina pessoal usando GitBash.
3 -> Na máquina de trabalho eu baixei o projeto que criei no GitHub usando a outra máquina.
4 -> Fiz algumas alterações no projeto e quero atualizar no GitHub .

Resumindo: Eu quero baixar meu próprio projeto do GitHub em outra máquina e salvar a atualização realizada.

Podem me ajudar?
vlw!

**Comment by felquis** [comment, marked as answer]
Essa é justamente a mágica do GIT mano o/

Após criar o repositório no GitHub, em cada computador, vc vai fazer um `git clone`. Caso seja um repositório privado, vc terá que cadastrar chaves ssh de cada computador no GitHub (Ex: estou presumindo que vc usa o mesmo usuário do GitHub em 2 computadores, então nessa conta vc terá 2 chaves ssh salvas no GitHub.)

A sincronização do código entre os computadores não acontece automáticamente como um dropbox, vc terá que usar comandos como:

No computador 1, vc usa os três comandos `git add`, `git commit` e `git push` para enviar arquivos modificados para o servidor.

No computador 2, vc faz `git fetch` e `git pull` para trazer atualizações do servidor.

Espero que outras pessoas possam passar o link de tutoriais mais bem ilustrados e simplificados para ajudar tbm. Qualquer dúvida manda ai nos comentários.

**Reply by HelenOliveira** [reply]
E se eu uso e-mails diferentes (username e email) em cada computador? Fica do mesmo jeito?

**Reply by felquis** [reply]
Oi @HelenOliveira Helen, tudo bem? O processo é o mesmo, a dúvida inicial parece ser até uma pegadinha para iniciantes.

Qualquer maquina que for clonar um projeto GIT privado precisa ter sua chave publica SSH cadastrada no servidor.

Existem configurações um pouco mais avançadas onde vc pode ter em 1 só maquina várias chaves SSH - mas creio que para simplificar para iniciantes em geral, 1 maquina é igual a sua própria chave SSH publica salva no servidor que hospeda o repositório.

Espero ter te ajudado, qualquer dúvida crie uma nova discussão por favor 😄

**Comment by felquis** [comment]
3 -> Na máquina de trabalho eu baixei o projeto que criei no GitHub usando a outra máquina.

Esse processo de "baixar" em outra maquina, a gente chama de clone.

4 -> Fiz algumas alterações no projeto e quero atualizar no GitHub .

Depois de fazer o clone, vc só vai conseguir fazer push de alterações para o servidor se vc tiver com a chave ssh configurada corretamente no computador, https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Essa chave, vc deve pegar a publica dela, e salvar no seu profile do GitHub https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account

Veja se isso te ajuda

**Comment by wilsonneto-swe** [comment]
Opa @ValdirCezar , assim como o @felquis disse, trabalho em várias máquinas também e é tranquilo, clona teu projeto na máquina, faz as alterações, commita e manda o push de volta pro github... Nas outras máquinas basta dar um pull que vai ir a alteração e assim continuar trabalhando, brincando ou estudando onde estiver.

Forte abraço!

**Comment by ValdirCezar** [comment]
Obrigado pessoal!
